Code Review

Chaoskil opened the pull request:
lets see if this works...

Chaoskil committed:
re-adding scripts for code review

TheRealNigl commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,62 @@
+
+    //variable that needs to be applied within unity
+    [SerializeField]
+    private float inactivatedScale = 1, activatedScale = 1.5f;
```
Either assign the variables in the Start Function or serialize the field. I recommend not serializing a field.

TheRealNigl commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,62 @@
+
+
+    //other variables that dont need to be applied in Unity
+    private bool isActivated = false;
```
Assign variables in the Start function

TheRealNigl commented on Assets/Scripts/Checkpoint.cs:
```diff
@@ -0,0 +1,62 @@
+    {
+        if (collision.CompareTag("Player") && !isActivated)
+        {
+            RobotControllerScript player = collision.GetComponent<RobotControllerScript>();
```
A simpler statement is "collision.GetComponent<RobotControllerScript>().SetCurrentCheckpoint(this);"

Chaoskil replied:
except it doesn't work when I replace it with what your suggesting...

Chaoskil replied:
needed the <RobotControllerScript> with it.

TheRealNigl commented on Assets/Scripts/Collectibles.cs:
```diff
@@ -0,0 +1,45 @@
+
+    //variable that needs to be set inside of Unity
+    [SerializeField]
+    private Animator anim;
```
GetComponent<Animator>(); in Start?

Chaoskil replied:
"GetComponent<>();" please specify the exact way to use it.

TheRealNigl commented on Assets/Scripts/Collectibles.cs:
```diff
@@ -0,0 +1,45 @@
+            anim.SetBool("Collected", true);
+            boxCollider2D.enabled = false;
+            coinCount++;
+            Debug.Log("Coin count: " + coinCount);
```
Please remove Debug.log

TheRealNigl commented on Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,66 @@
+    private string sceneToLoad;
+
+    [SerializeField]
+    private Animator anim;
```
GetComponent<Animator>(); in Start?

TheRealNigl commented on Assets/Scripts/Door.cs:
```diff
@@ -0,0 +1,66 @@
+
+    //variables that do not need to be set within Unity
+    private bool isPlayerInTrigger;
+    private bool openDoor = false;
```
Initialization in Start?

TheRealNigl commented on Assets/Scripts/Keys.cs:
```diff
@@ -0,0 +1,48 @@
+
+            //turns off the keys hitbox and sends a message saying that the player has the key, plays the key sound as well
+            boxCollider2D.enabled = false;
+            Debug.Log("You have the key!");
```
Please remove debug.log

TheRealNigl commented on Assets/Scripts/Keys.cs:
```diff
@@ -0,0 +1,48 @@
+        {
+            //gets access to the players script and tells it that the player has the key
+            RobotControllerScript player = collision.GetComponent<RobotControllerScript>();
+            player.DoesPlayerHaveKey(true);
```
collision.GetComponent<RobotControllerScript>().DoesPlayerHaveKey(true);

TheRealNigl commented on Assets/Scripts/RobotControllerScript.cs:
```diff
@@ -0,0 +1,204 @@
+    private int beforeRespawn = 5000;
+    private int timer = 0;
+    private AudioSource audioSource;
+    private bool pickUpKey = false;
```
Initializations in Start!!!

TheRealNigl commented on Assets/Scripts/RobotControllerScript.cs:
```diff
@@ -0,0 +1,204 @@
+    private PhysicsMaterial2D playerMovingPhysicsMaterial, playerStoppingPhysicsMaterial;
+
+    [SerializeField]
+    private Collider2D playerGroundCollider;
```
Consider getting the above Unity components with GetComponent<*SpecificComponent*>()

Chaoskil replied:
There's the way to write it, thank you.

Chaoskil replied:
changed as many as I could about 4 - 5 of them i think.

TheRealNigl reviewed (commented):
Please take a look at the comments. Let me know your thoughts

Chaoskil committed:
code review p2

- added in all changes from the first run through by reviewer

Chaoskil commented:
fixed everything (i think) that was commented on.

Chaoskil committed:
Level 1 complete

- built all of level 1 and tested it a bit, doesnt seem to have problems
- added an extra sprite for an underground door (helps make the levels make a bit of sense)

Chaoskil committed:
Level 2 (half)

- drew the layout for level 2
- painted the background for it
- have most of the colliders worked out
- have the player in the level

Chaoskil committed:
level 2 complete

- finished adding colliders to everything that needed them
- placed collectibles
- placed checkpoints
- set up hazards and platforms
-set up exit door and key for it
adjusted player script slightly for better ground detection

Chaoskil committed:
Level 3 (part 1)

- created the very basic layout of what it should look like
- added the player and camera
- filled in most of the background

Chaoskil committed:
Level 3 Complete

- finished painting the level
- spread out collectibles
- made platforms
- spread out checkpoints
- placed key
- placed hazards
- set up end point

Chaoskil committed:
Start and Credits

- Finished both the start and credits screens
- created backgrounds for both screens to make them stand out
- used some assets from the menu tutorial by Unity (gave credit)
- Adjusted fonts to fit game better
- adjusted some physics in level 2

Chaoskil committed:
script comments

Forget to put the credit comment to the other Unity tutorial scripts, fixed that here

TheRealNigl commented:
Sorry for not saying to accept. I just saw the email for the merged branch. You have good coding practices, no overused variables, and a good focus on functions for class. I would recommend to go for more private variables, then use resources.load, get file text, vectors, or floats assignment for the variable initialization. Having a fluid design between classes is important, which I mostly see, finding your style will help you convey the information.
